src/facility.h

```diff
@@ -161,6 +161,8 @@ class Facility : public TimeListener, public Agent, public Trader {
       const std::vector<std::pair<Trade<Product>,
       Product::Ptr> >& responses) {}
 
+  virtual bool IsShim() { return false; }
```

Comment:
I'm not 100% clear on the inheritance between the C++ and the python, but if this was implemented as a default implementation in `time_listener.h` wouldn't it prevent the need from implementing it in each of `facility.h`, `institution.h` and `region.h`?  Or does the python agent somehow inherit from that, too?

Reply:
Good point, `IsShim()` is now only implemented in `time_listener.h`